nntrainer/optimizers/adam.cpp

```diff
@@ -26,17 +26,12 @@ const std::string Adam::type = "adam";
 
 enum AdamParams { wm, wv };
 
-void Adam::addOptimizerVariable(std::vector<Weight> &weight_list) {
-  for (auto &w : weight_list) {
-    w.clearOptimizerVariables();
+std::vector<TensorDim> Adam::addOptimizerVariable(const TensorDim &dim) {
+  std::vector<TensorDim> opt_vars_dim;
+  opt_vars_dim.push_back(dim); /** Add wm */
+  opt_vars_dim.push_back(dim); /** Add wv */
 
-    // TODO: only trainable weights must be sent to optimizer
-    if (!w.getTrainable())
-      continue;
-
-    w.addOptimizerVariable(w.getDim()); /** Add wm */
-    w.addOptimizerVariable(w.getDim()); /** Add wv */
-  }
+  return opt_vars_dim;
```

Comment:
Super minor suggestion, how about return `{dim, dim}` ?

Reply:
That would be much clear than this way thanks